docs/docs/reference/built-in-components/gatsby-plugin-image.md

```diff
@@ -0,0 +1,129 @@
+
+> It is a very good idea to use [the GraphiQL IDE](/docs/how-to/querying-data/running-queries-with-graphiql) when writing your `gatsbyImageData` queries. It includes auto-complete and inline documentation for all of the options and lets you see the generated image data right inside the IDE.
+
+### Layout
+
+**Prop name: `layout`**
+
+The image components support three types of layout, which determine the image sizes that are generated, as well as the resizing behavior of the image itself in the browser. You can compare the layouts in the video below:
+
+<video controls autoplay loop>
+  <source type="video/mp4" src="./layouts.mp4" />
+  <p>Your browser does not support the video element.</p>
+</video>
+
+To set the layout, pass in the type to the layout prop. e.g.
+
+```jsx
+<StaticImage
+  src="./dino.png"
+  alt="A dinosaur"
+  // highlight-next-line
+  layout="fixed"
+/>
+```
+
+Alternatively, pass it to the resolver:
+
+```graphql
+dino {
+  childImageSharp {
+    # highlight-next-line
+    gatsbyImageData(layout: FIXED)
+  }
+}
+```
+
+#### Constrained
+
+_Component prop: `"constrained"`. Resolver prop: `CONSTRAINED`_
+
+This is the default layout. It displays the image at the size of the source image, or you can set a maximum size by passing in `width` or `height`). If the screen or container size is less than the width of the image, it scales down to fit, maintaining its aspect ratio. It generates smaller versions of the image so that a mobile browser doesn't need to load the full-size image.
+
+#### Fixed
+
+_Component prop: `"fixed"`. Resolver prop: `FIXED`_
+
+This is a fixed-size image. It will always display at the same size, and will not shrink to fit its container. This is either the size of the source image, or the size set by the `width` and `height` props. Only use this if you are certain that the container will never need to be narrower than the image.
+
+#### Full Width
+
+_Component prop: `"fullWidth"`. Resolver prop: `FULL_WIDTH`_
+
+Use this for images that are always displayed at the full width of the screen, such as banners or hero images. Like the constrained layout, this resizes to fit the container. However it is not restricted to a maximum size, so will grow to fill the container however large it is, maintaining its aspect ratio. It generates several smaller image sizes for different screen breakpoints, so that the browser only needs to load one large enough to fit the screen. You can pass a `breakpoints` prop if you want to specify the sizes to use, though in most cases you can allow it to use the default.
+
+### Sizing options
```

Comment:
For consistency, I'd love it if all the prop names could be at the same heading level. (vs. right now there's a difference between the two sizing options and the other three.)

How would you feel about getting rid of this "Sizing options" level and just having two separate h3s for `width / height` and `aspectRatio`?

Reply:
If we go down that route, the note on line 73 ("Size props are optional...") could maybe go in the `width / height` section? Or combined with the paragraph on line 81 ("If you set just one of these...")?